pkg/ibmvpc/sg_analysis.go

```diff
@@ -317,6 +317,26 @@ func AnalyzeSGRules(rules []*SGRule, isIngress bool) *ConnectivityResult {
 	return res
 }
 
+func mapAndAnalyzeSGRules(rules []*SGRule, isIngress bool, connectivityMap map[*ipblock.IPBlock]*ConnectivityResult) {
+	Locals := []*ipblock.IPBlock{}
+	for i := range rules {
+		if rules[i].local != nil {
+			Locals = append(Locals, rules[i].local)
+		}
+	}
+	disjointLocals := ipblock.DisjointIPBlocks(Locals, []*ipblock.IPBlock{ipblock.GetCidrAll()})
+	for i := range disjointLocals {
+		relevantRules := []*SGRule{}
+		for j := range rules {
+			if disjointLocals[i].ContainedIn(rules[j].local) {
+				relevantRules = append(relevantRules, rules[j])
+			}
+		}
+		// check if we already called AnalyzeSGRules with the same relevantRules
```

Comment:
is this comment still a TODO? 
should also be handled.. (caching and reusing computed connectivity results for same set of rules) .
consider checking `pkg/common/genericSet.go` for generating key per set of rules .

Reply:
also make sure there is a relevant test with caching scenario after this is added.

Reply:
The test I added has a cache scenario

Reply:
can you add a test to validate that caching actually worked on that scenario? currently that test could pass even if caching did not work from some reason.